### ISSUE TITLE
[Feature Request]: Examples on RAGFlow API

### ISSUE THREAD

**Issue by JinHai-CN** [infiniflow/ragflow#4310]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Describe the feature you'd like

Although there is already documentation for the RAGFlow API, it would be easier for developers to use if specific examples were provided. This issue proposes creating examples for the RAGFlow SDK, and accessing HTTP interfaces with cURL. 

Python SDK example: example/sdk
HTTP API with cURL: example/http

**Comment by JinHai-CN** [comment]
dataset related examples:
https://github.com/infiniflow/ragflow/blob/main/example/sdk/dataset_example.py
https://github.com/infiniflow/ragflow/blob/main/example/http/dataset_example.sh

**Comment by saishreyakumar** [comment]
Hi @simin75simin @KevinHuSh , just checking in, is this issue still active or are you currently working on it? I would love to help out if it's available. Thanks!

**Comment by simin75simin** [comment]
> Hi [@simin75simin](https://github.com/simin75simin) , just checking in, is this issue still active or are you currently working on it? I would love to help out if it's available. Thanks!

yeah, i think i'd get a pr in a few days, sorry I was working on other things before, i was also checking with other people on discord couple weeks ago regarding this once i got some time.

**Comment by bulolo** [comment]
is there any way  chat/completions only by a file? llike this

```
{
        model: 'glm-4-flash',
        messages: formattedMessages,
        stream: true,
        temperature: 0.7,
        max_tokens: 9000,
        files: [
          {
            type: "file",
            id: "82b8653b-3399-4edb-97f6-34e9b4e7df67"
          },
          {
            type: "collection",
            id: "fcf35088-7f8a-47ed-9a2f-a410e0390d43"
          }
        ]
      }
```

**Comment by simin75simin** [comment]
> is there any way chat/completions only by a file? llike this
> 
> ```
> {
>         model: 'glm-4-flash',
>         messages: formattedMessages,
>         stream: true,
>         temperature: 0.7,
>         max_tokens: 9000,
>         files: [
>           {
>             type: "file",
>             id: "82b8653b-3399-4edb-97f6-34e9b4e7df67"
>           },
>           {
>             type: "collection",
>             id: "fcf35088-7f8a-47ed-9a2f-a410e0390d43"
>           }
>         ]
>       }
> ```

not sure what you mean?

**Comment by bulolo** [comment]
> > is there any way chat/completions only by a file? llike this
> > 
> > ```
> > {
> >         model: 'glm-4-flash',
> >         messages: formattedMessages,
> >         stream: true,
> >         temperature: 0.7,
> >         max_tokens: 9000,
> >         files: [
> >           {
> >             type: "file",
> >             id: "82b8653b-3399-4edb-97f6-34e9b4e7df67"
> >           },
> >           {
> >             type: "collection",
> >             id: "fcf35088-7f8a-47ed-9a2f-a410e0390d43"
> >           }
> >         ]
> >       }
> > ```
> 
> not sure what you mean?

api to talk with Specify file

**Comment by simin75simin** [comment]
> > > is there any way chat/completions only by a file? llike this
> > > ```
> > > {
> > >         model: 'glm-4-flash',
> > >         messages: formattedMessages,
> > >         stream: true,
> > >         temperature: 0.7,
> > >         max_tokens: 9000,
> > >         files: [
> > >           {
> > >             type: "file",
> > >             id: "82b8653b-3399-4edb-97f6-34e9b4e7df67"
> > >           },
> > >           {
> > >             type: "collection",
> > >             id: "fcf35088-7f8a-47ed-9a2f-a410e0390d43"
> > >           }
> > >         ]
> > >       }
> > > ```
> > 
> > 
> > not sure what you mean?
> 
> api to talk with Specify file

i'd say no at this point but if you know better pls let me know

**Comment by bulolo** [comment]
english:
i would better to  support this api [chat/completions](https://ragflow.io/docs/dev/http_api_reference#create-chat-completion) with files params to chat with a Specify file

like open webui 's api 
https://docs.openwebui.com/getting-started/api-endpoints/#using-files-and-collections-in-chat-completions

![Image](https://github.com/user-attachments/assets/50fd6f8f-6784-40ce-8b35-908372be686a)

@simin75simin 


or how to make a chat/completions to chat with a Specify file


中文：

现在ragflow api并没有提供一个能针对知识库中某个文件的对话api参数，是否支持，或者时候已经支持？

例如 像open webui的这种针对某个文件
![Image](https://github.com/user-attachments/assets/e5d91cb0-2cd5-4499-975e-1172b5825ea0)

**Comment by simin75simin** [comment]
> english: i would better to support this api [chat/completions](https://ragflow.io/docs/dev/http_api_reference#create-chat-completion) with files params to chat with a Specify file
> 
> like open webui 's api https://docs.openwebui.com/getting-started/api-endpoints/#using-files-and-collections-in-chat-completions
> 
> ![Image](https://github.com/user-attachments/assets/50fd6f8f-6784-40ce-8b35-908372be686a)
> 
> [@simin75simin](https://github.com/simin75simin)
> 
> or how to make a chat/completions to chat with a Specify file
> 
> 中文：
> 
> 现在ragflow api并没有提供一个能针对知识库中某个文件的对话api参数，是否支持，或者时候已经支持？
> 
> 例如 像open webui的这种针对某个文件 ![Image](https://github.com/user-attachments/assets/e5d91cb0-2cd5-4499-975e-1172b5825ea0)

actually pls stop discussing this here as it's not relevant to the issue. consult the documentation for specific functionalities.

**Comment by simin75simin** [comment]
> Hi [@simin75simin](https://github.com/simin75simin) [@KevinHuSh](https://github.com/KevinHuSh) , just checking in, is this issue still active or are you currently working on it? I would love to help out if it's available. Thanks!

actually i think it's just porting the various API scripts from say ragflow/blob/main/example/http to https://ragflow.io/docs/dev/http_api_reference#converse-with-agent and do the same for Python APIs..

correct me if i'm wrong....

**Comment by wwwqqyy** [comment]
I want to create an example that includes reading a file, chunking, extracting entities and relationships, conducting query recall, and generating the final answer. I think this is very beneficial for developers.

**Comment by tysoncung** [comment]
I'm interested in working on this issue. Could you provide more context about what you're looking for? Any additional details about requirements or constraints would be helpful.

**Comment by Br1an67** [comment]
Hi @simin75simin — are you still working on this? I noticed it's been a while since the last update. If not, I'd be happy to pick it up.

I've added SDK and cURL examples for chat, document, and retrieval APIs to complement the existing dataset examples.

PR: #13284
